### PR TITLE
Add .NET Standard 2.0 methods/properties to NetworkStream

### PR DESCRIPTION
**Pull request by ericeil**
Part of #11788.

@stephentoub @CIPop @davidsh

### CODE REVIEW

**Comment by ericeil** [comment]
Filed #11904 to cover adding more testing to NetworkStream in general.

File: src/System.Net.Sockets/ref/System.Net.Sockets.cs [hunk 1]
```diff
@@ -88,17 +88,27 @@ public partial class NetworkStream : System.IO.Stream
...
         public override bool CanWrite { get { return default(bool); } }
         public virtual bool DataAvailable { get { return default(bool); } }
         public override long Length { get { return default(long); } }
         public override long Position { get { return default(long); } set { } }
+        protected bool Readable { get { return default(bool); } set { } }
         public override int ReadTimeout { get { return default(int); } set { } }
+        protected Socket Socket { get { return default(Socket); } }
+        protected bool Writeable { get { return default(bool); } set { } }
         public override int WriteTimeout { get { return default(int); } set { } }
+        public IAsyncResult BeginRead(byte[] buffer, int offset, int size, AsyncCallback callback, Object state) { return default(IAsyncResult); }
```
Comment (CIPop) [inline src/System.Net.Sockets/ref/System.Net.Sockets.cs]: I remember we had testing for APM that got removed at some point. (Didn't look - maybe the code is still there?).
The tests should be in older versions of the repo and can easily be brought back (I believe they were in the System.Net.Sockets.Legacy folder.)

File: src/System.Net.Sockets/ref/System.Net.Sockets.cs [hunk 2]
```diff
@@ -88,17 +88,27 @@ public partial class NetworkStream : System.IO.Stream
...
+        public IAsyncResult BeginWrite(byte[] buffer, int offset, int size, AsyncCallback callback, Object state) { return default(IAsyncResult); }
+        public void Close(int timeout) { }
```
Comment (CIPop) [inline src/System.Net.Sockets/ref/System.Net.Sockets.cs]: Please add a TODO / issue to track testing for 100% API coverage (at least one test calling into the API, not necessarily all scenarios in the first phase.)

File: src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs [hunk 3]
```diff
@@ -70,10 +70,42 @@ internal NetworkStream(NetworkStream networkStream, bool ownsSocket)
...
+                if (socket == null)
+                {
+                    throw new ArgumentNullException(nameof(socket));
+                }
+                InitNetworkStream(socket, access);
+#if DEBUG
+            }
+#endif
+        }
+        public NetworkStream(Socket socket, FileAccess access, bool ownsSocket)
```
Comment (stephentoub) [inline src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs]: Could this just be:
```C#
public NetworkStream(Socket socket, FileAccess access, bool ownsSocket) : this(socket, access)
{
    _ownsSocket = ownsSocket;
}
```
?

Comment (stephentoub) [inline src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs]: (Or some other chaining of the ctors?  There seems to be lots of duplication.)

**Comment by ericeil** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test please (https://github.com/dotnet/corefx/issues/11908)

File: src/System.Net.Sockets/ref/System.Net.Sockets.cs (hunk 1 above)
Comment (ericeil) [inline src/System.Net.Sockets/ref/System.Net.Sockets.cs]: Thanks!  I added this note to #11904.

File: src/System.Net.Sockets/ref/System.Net.Sockets.cs (hunk 2 above)
Comment (ericeil) [inline src/System.Net.Sockets/ref/System.Net.Sockets.cs]: Tracked by #11904

File: src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs (hunk 3 above)
Comment (ericeil) [inline src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs]: Indeed.  Added a commit that consolidates the constructor logic.

**Comment by ericeil** [comment]
[Another CentOS segfault](http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/master/job/centos7.1_release_prtest/409/)

**Pull request merged**